lib/utilities/attrValidators.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * More validators for class attributes. 
+ * @class AttrValidators
+ * @static
+ */
+var AttrValidators = {
+
+  /**
+   * Returns a function that checks if a value is an instance of the given
+   * contructor isInstanceOf.
+   *
+   * @method isInstanceOf
+   * @param ctor {Object} The contructor function
+   */
+  isInstanceOf: function(ctor) {
+    if (!ctor) {
+      console.error(ctor);
+      console.error('Constructor object to check against was falsy.');
+      return false;
+    }
+
+    return function (val, attrName) {
+      return (val instanceof ctor);
+    };
+  },
+
+  /**
+   * Returns a function that checks if a value is in the given constants object.
+   *
+   * @method isValueIn
+   * @param obj {Object} The object of constants (key integer pairs)
+   */
+  isValueIn: function (obj) {
+    if (!obj) {
+      console.error(obj);
+      console.error('Constants object to check against was falsy.');
+      return false;
+    }
+    
+    return function (val, attrName) {
+      var values = [];
+
+      for (var key in obj) {
+        values.push(obj[key]);
+      }
+
+      return values.indexOf(val) !== -1;
```

Comment:
Very true. It looks like I opened my function too early. If I just move line 40 down to line 46 it will make a lot more sense. 
I can make it more concise with `.key.map` but Javascript has no faster way to get all values.